tests/test_views.py

```diff
@@ -0,0 +1,128 @@
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1,
+                month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+        )
+        post = Post.objects.create(
+            author=self.user, title='Test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=3,
+                year=2020, tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=3,
+                year=2020,
+                tzinfo=time_zone,
+            ),
+        )
+        past_post = Post.objects.create(
+            author=self.user, title='past_est',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+        )
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=1,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [past_post])
+            self.assertNotContains(response, post)
+            self.assertNotContains(response, future_post)
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=10,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(
+                list(response.context['posts']), [
+                    post,
+                    past_post,
+                ],
+            )
+            self.assertNotContains(response, future_post)
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=4,
+                    year=3020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(
+                list(response.context['posts']), [
+                    future_post,
+                    post,
+                    past_post,
+                ],
+            )
+        response = self.client.get(reverse('post_list'))
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed(response, 'blog/post_list.html')
+        self.assertNotContains(response, future_post)
```

Comment:
This will fail when this test is run in the year 3021. This is because you don't have a deterministic fixed date in this last block. I'm not sure if you really need it but if you want, you could also test this with some date back in 2018.